src/main/java/org/terasology/web/resources/games/NewGameAction.java

```diff
@@ -0,0 +1,101 @@
+import org.terasology.web.resources.ResourceAccessException;
+import org.terasology.world.internal.WorldInfo;
+import org.terasology.world.time.WorldTime;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.terasology.web.resources.InputCheckUtils.checkNotNull;
+import static org.terasology.web.resources.InputCheckUtils.checkNotNullOrEmpty;
+
+public class NewGameAction extends AbstractAction {
+
+    private String gameName;
+    private String seed;
+    private List<Name> modules;
+    private SimpleUri worldGenerator;
+
+    private transient DependencyResolver dependencyResolver;
+
+    @Override
+    public void perform(ModuleManager moduleManager) throws ResourceAccessException {
+        checkNotNullOrEmpty(gameName, "A name for the new game must be specified.");
+        checkNotNullOrEmpty(seed, "A seed must be specified.");
+        checkNotNull(modules, "A list of modules must be specified");
+        checkNotNull(worldGenerator, "A world generator must be specified.");
+        if (Files.exists(getPathManager().getSavePath(gameName))) {
+            throw new ResourceAccessException(new ActionResult(ActionResult.Status.CONFLICT, "A game with the specified name already exists"));
+        }
+        if (dependencyResolver == null) {
+            dependencyResolver = new DependencyResolver(moduleManager.getRegistry());
+        }
+        GameManifest newGameManifest = buildGameManifest();
+        Path saveDir = getPathManager().getSavePath(newGameManifest.getTitle());
+        Path saveFile = saveDir.resolve(GameManifest.DEFAULT_FILE_NAME);
+        try {
+            Files.createDirectory(saveDir);
+            GameManifest.save(saveFile, newGameManifest);
+        } catch (IOException ex) {
+            throw new ResourceAccessException(new ActionResult(ActionResult.Status.GENERIC_ERROR, "Failed to save new game manifest: " + ex.getMessage()));
+        }
+    }
+
+    private GameManifest buildGameManifest() throws ResourceAccessException {
+        GameManifest newGameManifest = new GameManifest();
+        newGameManifest.setTitle(gameName);
+        newGameManifest.setSeed(seed);
+
+        ResolutionResult result = dependencyResolver.resolve(modules);
+        if (!result.isSuccess()) {
+            throw new ResourceAccessException(new ActionResult(ActionResult.Status.GENERIC_ERROR, "Failed to resolve all module dependencies"));
+        }
+        for (Module m: result.getModules()) {
+            newGameManifest.addModule(m.getId(), m.getVersion());
+        }
+
+        WorldInfo worldInfo = new WorldInfo(TerasologyConstants.MAIN_WORLD, newGameManifest.getSeed(),
+                (long) (WorldTime.DAY_LENGTH * 0.025f), worldGenerator);
+        newGameManifest.addWorld(worldInfo);
+
+        return newGameManifest;
+    }
+
+    void setFields(String fGameName, String fSeed, List<Name> fModules, SimpleUri fWorldGenerator) {
```

Comment:
Looks like something that is useful for testing. Would it make sense to convert this into a package private constructor?